src/main/java/seedu/address/logic/commands/ViewCommand.java

```diff
@@ -0,0 +1,75 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ACTIVITY_ID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CONTACT_ID;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.model.Model;
+import seedu.address.model.activity.Activity;
+import seedu.address.model.person.Person;
+
+/**
+ * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class ViewCommand extends Command {
+
+    public static final String COMMAND_WORD = "view";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds either an activity or contact with "
+            + "the specified ID number and displays them. ID must be numerical!\n" + "Parameters: {"
+            + PREFIX_ACTIVITY_ID + " | " + PREFIX_CONTACT_ID + "}ID\n"
+            + "Example: " + COMMAND_WORD + "a/2";
+
+    private final int id;
+    private final Prefix prefix;
+
+    public ViewCommand(int id, Prefix prefix) {
+        this.id = id;
+        this.prefix = prefix;
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        Predicate<Person> predicate;
+        if (prefix.equals(PREFIX_ACTIVITY_ID)) {
+            Activity a = model.getActivityBook().getActivity(id);
+            if (a == null) {
+                return new CommandResult(
+                        String.format(Messages.MESSAGE_ACTIVITY_NONEXISTENT, id));
+            }
+            predicate = (Person x) -> a.hasPerson(x.getPrimaryKey());
+        } else if (prefix.equals(PREFIX_CONTACT_ID)) {
+            predicate = (Person x) -> x.getPrimaryKey() == id;
+        } else {
+            predicate = x -> true; // default to showing everyone if no args
+        }
+
+        model.updateFilteredPersonList(predicate, model);
```

Comment:
This is just a placeholder for viewing activities right? Like our actual activity viewing logic isn't going to be like this